src/hypothesis/strategies.py

```diff
@@ -982,32 +987,61 @@ def fractions(min_value=None, max_value=None, max_denominator=None):
 
     If max_denominator is not None then the absolute value of the denominator
     of any generated values is no greater than max_denominator. Note that
-    max_denominator must be at least 1.
+    max_denominator must be None or a positive integer, and the denominator
```

Comment:
I think we're probably spending a lot of time worrying about a niche case that doesn't matter all that much and it's OK to handle a bit suboptimally.

Here's a proposed simple solution:

1. We stop worrying about catching conflicts between bounds and `max_denominator` up front and allow unsatisfiable tests to fail at runtime.
2. We do this by going back to the proposed clipping behaviour - if when rounding we get something outside the bounds, we snap to the nearest bound.
3. If the bound in question has too large a denominator, we filter out the example with assume and allow example generation to try again.

I think that handles all the happy paths well and degrades gracefully in the presence of hard or impossible to satisfy constraints.

How does that sound?